tutorials/01-gaussian-mixture-model/01_gaussian-mixture-model.jmd

```diff
@@ -186,8 +242,168 @@ scatter(
 )
 ```
 
-```julia, echo=false, skip="notebook", tangle=false
+## Marginalizing Out The Assignments
+
+We can write out the marginal posterior of (continuous) $w, \mu$ by summing out the influence of our (discrete) assignments $z_i$ from 
+our likelihood:
+
+$$
+p(y \mid w, \mu ) = \sum_{k=1}^K w_k p_k(y \mid \mu_k)
+$$
+
+In our case, this gives us:
+
+$$
+p(y \mid w, \mu) = \sum_{k=1}^K w_k \cdot \operatorname{MvNormal}(y \mid \mu_k, I)
+$$
+
+
+### Marginalizing By Hand
+
+We can implement the above version of the Gaussian mixture model in Turing as follows:
+
+First, Turing uses log-probabilities, so the likelihood above must be converted into log-space:
+
+$$
+\log \left( p(y \mid w, \mu) \right) = \text{logsumexp} \left[\log (w_k) + \log(\operatorname{MvNormal}(y \mid \mu_k, I)) \right]
+$$
+
+Where we sum the components with `logsumexp` from the [`StatsFuns.jl` package](https://github.com/JuliaStats/StatsFuns.jl).
+
+
+The manually incremented likelihood can be added to the log-probability with `Turing.@addlogprob!`, giving us the following model:
```

Comment:
IMO we should not recommend the use of `Turing.@addlogprob!` in it's so easy to misuse and to get (silently) wrong results because it operates completely outside of the `~` logic in Turing/DynamicPPL. Instead I think usually one should use `~` with a (possibly custom) distribution.

Reply:
Sounds good! I initially wasn't going to include that section for basically the reasons you bring up, but I ended up including it (even though I don't actually sample from that model) to motivate what's going on with the MixtureModel lpdf.

I can replace it with a custom distribution (although this might be a little long for a model that's really just exposition), or omit it entirely.